src/wp-content/themes/twentyfifteen/js/functions.js

```diff
@@ -5,130 +5,162 @@
+		Array.from( container.querySelectorAll( '.menu-item-has-children > a' ) ).forEach( function( menuLink ) {
+			var dropdownToggle = document.createElement( 'button' );
+			dropdownToggle.className = 'dropdown-toggle';
+			dropdownToggle.setAttribute( 'aria-expanded', 'false' );
+			dropdownToggle.innerHTML = screenReaderText.expand;
+			menuLink.parentElement.insertBefore( dropdownToggle, menuLink.nextElementSibling );
+		} );
 
 		// Toggle buttons and submenu items with active children menu items.
-		container.find( '.current-menu-ancestor > button' ).addClass( 'toggle-on' );
-		container.find( '.current-menu-ancestor > .sub-menu' ).addClass( 'toggled-on' );
-
-		container.find( '.dropdown-toggle' ).on( 'click', function( e ) {
-			var _this = $( this );
-			e.preventDefault();
-			_this.toggleClass( 'toggle-on' );
-			_this.next( '.children, .sub-menu' ).toggleClass( 'toggled-on' );
-			_this.attr( 'aria-expanded', _this.attr( 'aria-expanded' ) === 'false' ? 'true' : 'false' );
-			_this.html( _this.html() === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand );
+		Array.from( container.querySelectorAll( '.current-menu-ancestor > button, .current-menu-ancestor > .sub-menu' ) ).forEach( function( subItem ) {
+			subItem.className += 'toggle-on';
+		} );
+
+		Array.from( container.querySelectorAll( '.dropdown-toggle' ) ).forEach( function( dropdownToggle ) {
+			dropdownToggle.addEventListener( 'click', function( e ) {
+				e.preventDefault();
+				toggleClass( this, 'toggle-on' );
+				if ( this.nextElementSibling.tagName === 'UL' || this.nextElementSibling.tagName === 'OL' ) {
+					toggleClass( this.nextElementSibling, 'toggled-on' );
+				}
+				this.setAttribute( 'aria-expanded', this.getAttribute( 'aria-expanded' ) === 'false' ? 'true' : 'false' );
+				this.innerHTML = this.innerHTML === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand;
+			} );
 		} );
 	}
-	initMainNavigation( $( '.main-navigation' ) );
+	initMainNavigation( document.querySelector( '.main-navigation' ) );
 
 	// Re-initialize the main navigation when it is updated, persisting any existing submenu expanded states.
-	$( document ).on( 'customize-preview-menu-refreshed', function( e, params ) {
-		if ( 'primary' === params.wpNavMenuArgs.theme_location ) {
-			initMainNavigation( params.newContainer );
-
-			// Re-sync expanded states from oldContainer.
-			params.oldContainer.find( '.dropdown-toggle.toggle-on' ).each(function() {
-				var containerId = $( this ).parent().prop( 'id' );
-				$( params.newContainer ).find( '#' + containerId + ' > .dropdown-toggle' ).triggerHandler( 'click' );
-			});
-		}
-	});
+	// This is only relevant for the Customize preview where jQuery is expected to be loaded.
+	if ( window.jQuery ) {
+		var $ = window.jQuery;
+		$( document ).on( 'customize-preview-menu-refreshed', function( e, params ) {
+			if ( 'primary' === params.wpNavMenuArgs.theme_location ) {
```

Comment:
I know this is existing code, but since we're rewriting it, I believe Yoda is only meant to be a thing in PHP code, at least going by [the equality examples in the JS Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#equality).